src/tabular-api-surface.ts

```diff
@@ -736,12 +759,25 @@ Please use the format 'prezzy' or '${instance.name}/tables/prezzy'.`);
         err = null;
       }
 
+      const mutationErrors = Array.from(mutationErrorsByEntryIndex.values());
       if (mutationErrorsByEntryIndex.size !== 0) {
-        const mutationErrors = Array.from(mutationErrorsByEntryIndex.values());
         callback(new PartialFailureError(mutationErrors, err));
         return;
       }
-
+      if (err) {
```

Comment:
Can we also add a unit test for this?

Reply:
Added.

Actually, something else I should flag is the code in the `mutationErrorsByEntryIndex.size !== 0` check. It seems odd to me that as soon as we see one entry level error that we don't send the grpc level error code anymore and we don't send an error for each pending entry.

Should we make changes so that the `mutationErrorsByEntryIndex.size !== 0` and the `mutationErrorsByEntryIndex.size === 0` cases send back more similar errors?

Reply:
I think the logic right now is fine. We want to keep the entry level error code as is.